instrumentation/apache-dbcp-2.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachedbcp/ApacheDbcpInstrumentationTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachedbcp;
+
+import io.opentelemetry.instrumentation.apachedbcp.AbstractApacheDbcpInstrumentationTest;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import java.lang.management.ManagementFactory;
+import javax.management.ObjectName;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class ApacheDbcpInstrumentationTest extends AbstractApacheDbcpInstrumentationTest {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  @Override
+  protected void configure(BasicDataSource dataSource, String dataSourceName) throws Exception {
+    dataSource.preRegister(
+        ManagementFactory.getPlatformMBeanServer(),
+        new ObjectName("io.opentelemetry.db:name=" + dataSourceName));
```

Comment:
we should probably document somewhere that capturing Apache DBCP metrics via javaagent requires the user to add this code

Reply:
Added a readme to the `javaagent` module explaining this. However, the auto-instrumentation is intended for use cases where all beans get automatically registered anyway, such as Spring Boot.